src/PerformanceTests/Common/Scenarios/BaseRunner.cs

```diff
@@ -214,4 +224,32 @@ protected virtual Task Setup()
         await Task.Delay(Settings.RunDuration).ConfigureAwait(false);
         Log.Info("Run duration expired.");
     }
-}
\ No newline at end of file
+
+    static Random random = new Random();
+    async Task RetryWithBackoff(Func<Task> action, CancellationToken token, string id, int maxAttempts)
+    {
+        while (true)
+        {
+            var attempts = 0;
+            try
+            {
+                await action()
+                    .ConfigureAwait(false);
+                return;
+            }
+            catch (Exception ex)
+            {
+                if (attempts > maxAttempts) throw new InvalidOperationException("Exhausted send retries.", ex);
+                double next;
+                lock (random) next = random.NextDouble();
+                next *= 0.2; // max 20% jitter
+                next += 1D;
+                next *= 100 * Math.Pow(2, ++attempts);
```

Comment:
that seems to yield quite long delays (of multiple seconds) easily. Is that the intention?

Reply:
Yes, that is intentional 100 ms, 200ms, 400ms, 800ms, 1600ms, 3200ms plus a random 0-20% jitter.

Reply:
Shall I add a code comment?

Reply:
>  100 ms, 200ms, 400ms, 800ms, 1600ms, 3200ms plus a random 0-20% jitter.

then shouldn't this be `attempts++` instead of `++attempts`?

Reply:
Let's see, attempts is 0, gets increased, Math.Path(2,1) results in 2, so yes, good catch @timbussmann